src/lib/selective-sync.js

```diff
@@ -0,0 +1,52 @@
+var debug = require('debug')('dat')
+var path = require('path')
+var EventEmitter = require('events').EventEmitter
+
+module.exports = function selectiveSync (state, bus) {
+  var archive = state.dat.archive
+  debug('sparse mode. downloading metadata')
+  var emitter = new EventEmitter()
+
+  function download (entry) {
+    debug('selected', entry)
+    archive.stat(entry, function (err, stat) {
+      if (err) return bus.emit('exit:error', err)
+      if (stat.isDirectory()) downloadDir(entry, stat)
+      if (stat.isFile()) downloadFile(entry, stat)
+    })
+  }
+
+  function downloadDir (dirname, stat) {
+    debug('downloading dir', dirname)
+    archive.readdir(dirname, function (err, entries) {
+      if (err) return bus.emit('exit:error', err)
+      entries.forEach(function (entry) {
+        emitter.emit('download', path.join(dirname, entry))
```

Comment:
The idea was we could also figure out when it was done and call `emitter.emit('end')` .. right now it just stays open and you have to watch the download speed